In the last few exercises you've learned everything there is to know about heights and weights of baseball players. Now it's time to dive into another sport: soccer.

You've contacted FIFA for some data and they handed you two lists. The lists are the following:

In [2]:
import pandas as pd
df_fifa = pd.read_csv('fifa.csv', index_col = 0)
heights = df_fifa['height'].tolist()
positions = df_fifa['position'].tolist()
print('positions =', positions[:5])
print('heights =', heights[:5])

positions = ['GK', 'M', 'A', 'D', 'M']
heights = [191, 184, 185, 180, 181]


Each element in the lists corresponds to a player. The first list, `positions`, contains strings representing each player's position. The possible positions are: `'GK'` (goalkeeper), `'M'` (midfield), `'A'` (attack) and `'D'` (defense). The second list, `heights`, contains integers representing the height of the player in cm. The first player in the lists is a goalkeeper and is pretty tall (191 cm).

You're fairly confident that the median height of goalkeepers is higher than that of other players on the soccer field. Some of your friends don't believe you, so you are determined to show them using the data you received from FIFA and your newly acquired Python skills.

**INSTRUCTIONS**

- Convert `heights` and `positions`, which are regular lists, to numpy arrays. Call them `np_heights` and `np_positions`.
- Extract all the heights of the goalkeepers. You can use a little trick here: use `np_positions == 'GK'` as an index for `np_heights`. Assign the result to `gk_heights`.
- Extract all the heights of all the other players. This time use `np_positions != 'GK'` as an index for `np_heights`. Assign the result to `other_heights`.
- Print out the median height of the goalkeepers using `np.median()`. Replace `None` with the correct code.
- Do the same for the other players. Print out their median height. Replace `None` with the correct code.

In [11]:
# heights and positions are available as lists

# Import numpy
import numpy as np
import pandas as pd
# Convert positions and heights to numpy arrays: np_positions, np_heights
df_fifa = pd.read_csv('fifa.csv', usecols = ['position','height'])

heights = df_fifa['height'].tolist()
positions = df_fifa['position'].tolist()

np_heights = np.array(heights)
np_positions = np.array(positions)

# Creating numpy arrays
np_heights = np.array(heights)
np_positions = np.array(positions)

# Extracting heights of the GKs and others
gk_heights = np_heights[np_positions == 'GK']
other_heights = np_heights[np_positions != 'GK']

print("Median height of goalkeepers: " + str(np.median(gk_heights)))

print("Median height of other players: " + str(np.median(other_heights)))

Median height of goalkeepers: 188.0
Median height of other players: 181.0


In [4]:
df_fifa = pd.read_csv('fifa.csv', usecols = ['position','height'])
print("Median height of goalkeepers:", df_fifa.fillna(0).loc[df_fifa["position"] == "GK", "height"].median())
print("Median height of other players:", df_fifa.fillna(0).loc[df_fifa["position"] != "GK", "height"].median())

Median height of goalkeepers: 188.0
Median height of other players: 181.0


In [5]:
суму стандарт выдхил голк вищ180 і мед зросту зах нижч 190

SyntaxError: invalid syntax (<ipython-input-5-de328dc54d93>, line 1)

In [17]:
print(np.std(np.array(df_fifa.loc[df_fifa["position"] == "GK", "height"].tolist()) \
             [np.array(df_fifa.loc[df_fifa["position"] == "GK", "height"].tolist()) >180]) + \
      np.array(df_fifa.loc[df_fifa["position"] == "D", "height"].tolist()) \
      [np.array(df_fifa.loc[df_fifa["position"] == "D", "height"].tolist()) < 190].mean())

print(np.std(np_heights[(np_positions == "GK") & (np_heights > 180)]) + \
      np_heights[(np_positions == "D") & (np_heights < 190)].mean())

185.87705787482588
185.87705787482588


In [6]:
6 на 9 половину 1 і третину 3 (n i m) (2x i 3x)

SyntaxError: invalid syntax (<ipython-input-6-1a6f22bf8e27>, line 1)